x/metadata/keeper/record.go

```diff
@@ -131,52 +143,83 @@ func (k Keeper) ValidateRecordUpdate(ctx sdk.Context, existing *types.Record, pr
 		if existing.Name != proposed.Name {
 			return fmt.Errorf("the Name field of records cannot be changed")
 		}
-		if !existing.SessionId.Equals(proposed.SessionId) || existing.Name != proposed.Name {
-			return fmt.Errorf("the SessionId field of records cannot be changed")
+		if !existing.SessionId.Equals(proposed.SessionId) {
+			// Make sure the original session parties are signers.
+			session, found := k.GetSession(ctx, existing.SessionId)
+			if !found {
+				return fmt.Errorf("original session %s not found for existing record", existing.SessionId)
+			}
+			if err := k.ValidateAllPartiesAreSigners(session.Parties, signers); err != nil {
+				return fmt.Errorf("missing signer from original session %s: %w", session.SessionId, err)
+			}
 		}
 		// The existing specification id might be empty for old stuff.
 		// And for now, we'll allow the proposed specification id to be missing and set it appropriately below.
 		// But if we've got both, make sure they didn't change.
 		if !existing.SpecificationId.Empty() && !proposed.SpecificationId.Empty() && !existing.SpecificationId.Equals(proposed.SpecificationId) {
 			return fmt.Errorf("the SpecificationId of records cannot be changed")
 		}
+		// Get the existing output hashes as both a slice and a map counting occurrences.
+		existingOutputHashes := make([]string, len(existing.Outputs))
+		existingOutputHashMap := map[string]int{}
+		for i, o := range existing.Outputs {
+			existingOutputHashes[i] = o.Hash
+			existingOutputHashMap[o.Hash]++
+		}
+		// Get the orig output hashes as a map counting occurrences.
+		origOutputHashMap := map[string]int{}
+		for _, h := range origOutputHashes {
+			origOutputHashMap[h]++
+		}
+		// Make sure that all the existing values are in the provided originals.
+		notInOrig := FindMissing(existingOutputHashes, origOutputHashes)
```

Comment:
this isn't correct unless the user is superseding all previous values (presuming a list)